src/playground/utils/codemirror-linter-extension.js

```diff
@@ -271,14 +271,29 @@ function assignKeys(actions) {
 
 function renderDiagnostic(view, diagnostic) {
     const element = document.createElement("div");
+    const root = ReactDOM.createRoot(element);
 
-    ReactDOM.render(
-        <Popup
-            message={diagnostic.message}
-            onFix={diagnostic.actions && (() => diagnostic.actions[0].apply(view, diagnostic.from, diagnostic.to))}
-            ruleName={diagnostic.source.replace("jshint:", "")}
-        />,
-        element
+    // clean up ruleName first
+    // see https://github.com/codemirror/lang-javascript/blob/main/src/eslint.ts#L51
+    let ruleName = "";
+
+    if (diagnostic.source === "jshint") {
+
+        // leave ruleName blank
+    } else {
+
+        // remove 'jshint:' in the beginning
+        ruleName = diagnostic.source.replace(/^jshint:/u, "");
```

Comment:
I feel we can remove the condition by using this regexp. WDYT?

```suggestion
        ruleName = diagnostic.source.replace(/^jshint:?/u, "");
```